src/ui/shared/WindowParam.ts

```diff
@@ -0,0 +1,26 @@
+export type AppMode = 'onboarding' | 'wallet';
+export type WindowType = 'popup' | 'tab' | 'dialog';
+export type WindowLayout = 'column' | 'page';
+
+export enum WindowParam {
+  windowType = 'windowType',
+  windowLayout = 'windowLayout',
+  appMode = 'appMode',
+}
+
+export function getWindowParam<T>(param: WindowParam, defaultValue?: T) {
+  const url = new URL(window.location.href);
+  return (url.searchParams.get(param) as T) || defaultValue;
+}
+
+export function setAppMode(params: URLSearchParams, value: AppMode) {
```

Comment:
Personally, I would probably  prefer one function to set all params. May be this would be easier for usage. However, not critical

Reply:
I refactored this part again, and now we have three contexts:

- `urlContext` (with a set method to set all params at once)
- `windowContext` (for querying info about the window type and layout)
- `appContext` (for global app mode info: `onboarding` or `wallet`)

I think it looks better now since `appMode` isn't really part of `windowContext` like it was before.

Reply:
Refactored, now we only have `UrlContext` + `setUrlContext`, so not relevant anymore